analysis/src/xml.rs

```diff
@@ -0,0 +1,715 @@
+                                Some("bitmask") => {
+                                    registry.bitmask_aliases.push(Alias::from_node(type_node));
+                                }
+                                Some("handle") => {
+                                    registry.handle_aliases.push(Alias::from_node(type_node));
+                                }
+                                Some("enum") => {
+                                    registry.enum_aliases.push(Alias::from_node(type_node));
+                                }
+                                Some("struct") => {
+                                    registry.struct_aliases.push(Alias::from_node(type_node));
+                                }
+                                _ => (),
+                            }
+                        } else {
+                            match type_node.attribute("category") {
+                                Some("basetype") => {
+                                    registry.basetypes.push(BaseType::from_node(type_node))
+                                }
+                                Some("bitmask") => registry
+                                    .bitmask_types
+                                    .push(BitMaskType::from_node(type_node)),
+                                Some("handle") => {
+                                    registry.handles.push(Handle::from_node(type_node))
+                                }
+                                Some("enum") => {
+                                    registry.enum_types.push(EnumType::from_node(type_node))
+                                }
+                                Some("funcpointer") => registry
+                                    .funcpointers
+                                    .push(FuncPointer::from_node(type_node)),
+                                Some("struct") => {
+                                    registry.structs.push(Structure::from_node(type_node, api))
+                                }
+                                Some("union") => {
+                                    registry.unions.push(Structure::from_node(type_node, api));
+                                }
+                                None => {
+                                    registry.externals.push(External::from_node(type_node));
+                                }
+                                _ => (),
+                            }
+                        }
+                    }
+                }
+                "enums" => match registry_child.attribute("type") {
+                    Some("enum") => registry.enums.push(Enum::from_node(registry_child, api)),
+                    Some("bitmask") => registry
+                        .bitmasks
+                        .push(BitMask::from_node(registry_child, api)),
+                    _ => {
+                        if matches!(registry_child.attribute("name"), Some("API Constants")) {
```

Comment:
No `==` with perhaps a `.as_deref()`?

Reply:
Let's change it if you prefer it this way.